DOCS/man/input.rst

```diff
@@ -3545,6 +3545,12 @@ Property list
     The player itself does not use any data in it (although some builtin scripts may).
     The property is not preserved across player restarts.
 
+    The following sub-paths are reserved for internal uses or have special semantics:
+    ``user-data/osc``, ``user-data/mpv``. Unless noted otherwise, the semantics of
+    any properties under these sub-paths can change at any time and may not be relied
+    upon, and writing to these properties may prevent builtin scripts from working
```

Comment:
Some of the 3rd party osc replacements (including my own, for full disclosure) write to `user-data/osc/margins` so that the console can read it and avoid gettting obstructed. E.g [in uosc](https://github.com/tomasklaen/uosc/blob/fe0d394435ad562176ae850175ebd5a083f0a6c2/src/uosc/main.lua#L501-L504).

Of course, this isn't documented, so there was no guarantee made, but to me it seems like we should document some of the `user-data/osc` properties instead of scaring people away from them. It allows scripts to query, for example the osc margins, in a generic way (as opposed to having a different path for different osc scripts).

Reply:
Agreed that 3rd party scripts can use/generate these properties, and it would be nicer to document what we can instead of making it a blackbox, but without any promise to keep it as stable API.

Reply:
> and it would be nicer to document what we can instead of making it a blackbox, but without any promise to keep it as stable API.

We cannot have both ways, if it is documented it has to be stable to some degree. Else what is the point of documenting it and encouraging the use if it can break at any time?

Reply:
> We cannot have both ways

Define "cannot".

We've had things documented and deprecated for years, or documented with a note that it's subject to change at any time.

So yes, it's suboptimal, but it is useful to have the docs of the current behavior, even if it might change at a later time. And once things change, we can change the docs too.

IMO it's better than blackbox, and also better than docs which we guarantee to try and keep unmodified, especially when we know in advance that it's not a gauarantee we want to give.

Not everything is black and white, and sometimes there's value also in grays.

That's not to say that this is the best course of action in this case, but it's worth considering IMO.

Reply:
> Documentation should be clear and unambiguous. There is not place for grays.

You know very well this is not the gray I meant.

Obviously docs should be clear and unambiguous. Feel free to quote me suggesting otherwise before you reply to things I haven't said, and please don't pick a fight over things which we agree on.

The gray was to document AND to not guarantee that it's never changing in the future.

We've been doing it for a long time for few things, and I think there's value in that, especially when we we're not actually changing it frequently or.. ever, but we still don't want to guarantee it'll stay forever.

You clearly dislike it, which is fine. You're entitled to your own opinion.